youtube_dl/extractor/facebook.py

```diff
@@ -509,3 +512,116 @@ def _real_extract(self, url):
         return self.url_result(
             compat_urllib_parse_unquote(self._match_id(url)),
             FacebookIE.ie_key())
+
+
+class FacebookUserIE(InfoExtractor):
+    _VALID_URL = r'(?P<url>https?://(?:[^/]+\.)?facebook\.com/(?:pg/)?(?P<id>[^/?#&]+))/videos(?!/[\w\.])'
+    IE_NAME = "facebook:user"
+
+    _TESTS = [{
+        # page
+        'url': 'https://www.facebook.com/Coca-Cola/videos/?ref=page_internal',
+        'info_dict': {
+            'id': 'Coca-Cola'
+        },
+        'playlist_mincount': 90,
+    }, {
+        # profile (requires login)
+        'url': 'https://www.facebook.com/zuck/videos',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        user_id = mobj.group('id')
+
+        page = self._download_webpage(
+            url, user_id, 'Downloading user webpage')
+        fb_url = self._html_search_meta(
+            'al:android:url', page, default=None)
+        fb_url_mobj = re.match(r'fb://(?P<type>page|profile)/(?P<id>\d+)', fb_url)
+        if not fb_url_mobj:
+            raise ExtractorError('Could not extract page ID', expected=False)
+        page_id = fb_url_mobj.group('id')
+        fb_dtsg_ag = self._search_regex(
+            r'"async_get_token":"([\w\-:]+)"',
+            page, 'fb_dtsg_ag', default=None)
+        pagelet_token = self._search_regex(
+            r'pagelet_token:"([\w\-]+)"',
+            page, 'pagelet_token', default=None)
+        collection_token = self._search_regex(
+            r'pagelet_timeline_app_collection_([\d:]+)',
+            page, 'collection_token', default=None)
+        cursor = None
+        entries = []
+
+        if fb_url_mobj.group('type') == 'page':
+            endpoint = 'PagesVideoHubVideoContainerPagelet'
+            data = {
+                'page': page_id
+            }
+            page_re = r'<td[^>]+>.+?<a href="(?P<url>[^"]+)".+?</td>'
+        elif fb_url_mobj.group('type') == 'profile':
+            if not (fb_dtsg_ag and pagelet_token and collection_token):
+                raise ExtractorError('You must be logged in to extract profile videos', expected=True)
```

Comment:
You technically can't login with this extractor apart from using cookies.

Reply:
I have amended the extractor error to reflect this